Source/Core/Core/IOS/Network/IP/Top.h

```diff
@@ -74,6 +76,29 @@ class NetIPTopDevice : public Device
   void Update() override;
 
 private:
+  // MSVC std::future default constructor workaround
+  struct DefIPCReply : public IPCReply
+  {
+    DefIPCReply() : IPCReply(0) {}
+    DefIPCReply(IPCReply reply) : IPCReply(reply) {}
+  };
+
+  struct AsyncTask
+  {
+    IOS::HLE::Request request;
+    std::packaged_task<std::future<DefIPCReply>()> task;
+    std::future<DefIPCReply> future;
+  };
+
+  template <typename Method, typename Request>
+  std::optional<IPCReply> LaunchAsyncTask(Method method, const Request& request)
+  {
+    std::packaged_task<std::future<DefIPCReply>()> task(
```

Comment:
This will launch a new thread for every async task, which seems kind of expensive (especially on Windows). Can we use a Common::WorkQueueThread instead?